File: triple_x.js
const prompt = require("prompt")

const playGame = (difficulty = 1) => {
  console.log("You're a secret agent breaking into a LEVEL " + difficulty + " server room\n")
  console.log("Your SuperHacker 2000 tells you the following information...\n")

  const a = Math.round(Math.random() + difficulty + difficulty)
  const b = Math.round(Math.random() + difficulty + difficulty)
  const c = Math.round(Math.random() + difficulty + difficulty)

  const sum = a + b + c;
  const product = a * b * c;

  console.log()
  console.log("+ There are three numbers in the code\n")
  console.log("+ The codes multiply to give " + product)
  console.log("+ The codes add-up to " + sum)

  console.log()
  console.log("Enter the three code numbers separated by a space\n")
  console.log()

 let hasWon = false

 prompt.start()
  // Process the numbers in the input
    prompt.get(["answer"], (err, result) => {

      const answerArr = result.answer.split(' ')

      const answerSum = answerArr.reduce((acc, cur) => Number(acc) + Number(cur), 0)
      const answerProduct = answerArr.reduce((acc, cur) => Number(acc) * Number(cur), 1)
      console.log("Sum", answerSum, "Product", answerProduct)
      
      
      if (answerSum === sum && answerProduct === product) {
        console.log("CONGRATULATIONS, YOU'RE IN. Now hurry before anyone finds you.\n")
        console.log("==============================================================\n")
        hasWon = true
      } else {
        console.log("BOOM :-( You failed, try again!\n")
      }
    })
}

playGame()

File: triple_x.cpp
#include <iostream>
#include <ctime>

void PlayGameAtDifficulty(int difficulty)
{
  std::cout << "You are a new developer working on building a LEVEL " << difficulty << " video game.\n";
  std::cout << "You get an unknown error with cryptic instructions on how to solve the error.\n";

  srand(time(NULL)); // Create new random sequence based on time of day
  const int a = rand() % difficulty + difficulty;
  const int b = rand() % difficulty + difficulty;
  const int c = rand() % difficulty + difficulty;

  const int sum = a + b + c;
  const int product = a * b * c;

  std::cout << std::endl;
  std::cout << "* There are three numbers in the code\n";
  std::cout << "* The codes multiply to give " << product << std::endl;
  std::cout << "* The codes add-up to " << sum << std::endl;

  std::cout << std::endl;
  std::cout << "Enter the three code numbers followed by x\n";
  std::cout << std::endl;

  int answer;
  int trialSum = 0;
  int trialProduct = 1; // Note required starting product

  // Process the numbers in the input
  while(std::cin >> answer)
  {
    trialSum += answer;
    trialProduct *= answer;
  }

  std::cout << std::endl;
  if (trialSum == sum && trialProduct == product)
  {
    std::cout << "CONGRATULATIONS, YOU'RE FIXED THE BUG. Now onto the next one...\n";
    std::cout << "==============================================================\n";
  }
  else
  {
    std::cout << "BOOM :-( that wasn't the answer. Let's keep working on this mf bug.\n";
  }

}

int main()
{
  int difficulty = 2;
  int maxDifficulty = 10;
  
  while(difficulty <= maxDifficulty)
  {
    PlayGameAtDifficulty(difficulty);
    std::cin.clear(); // Clears the failbit
    std::cin.ignore(); // Discards the buffer
    ++difficulty;
  }

  std::cout << "WOW - You're a master hacker!\n";
  return 0; // Exit with no error code
}